src/components/Banner/MainBanner/index.tsx

```diff
@@ -1,8 +1,55 @@
+import { useEffect, useRef, useState } from 'react';
+
+import 여행01 from '../../../assets/Banner/MainBanner/여행01.png';
+import 여행02 from '../../../assets/Banner/MainBanner/여행02.png';
+import 여행03 from '../../../assets/Banner/MainBanner/여행03.png';
+
+import styles from './MainBanner.module.scss';
+
+const images = [여행01, 여행02, 여행03];
+
 const MainBanner = () => {
+  const [current, setCurrent] = useState(0);
+  const timeoutRef = useRef<number | null>(null);
+
+  useEffect(() => {
+    timeoutRef.current = window.setTimeout(() => {
+      setCurrent((prev) => (prev + 1) % images.length);
+    }, 5000);
+
+    return () => {
+      if (timeoutRef.current) clearTimeout(timeoutRef.current);
+    };
+  }, [current]);
```

Comment:
dependency array에 `current`을 넣은걸로 보이는데요~!
`current`를 사용하지를 않고 있어서 빼는게 좋습니다.
`setCurrent`로 `current`값을 업데이트하면 `useEffect`가 다시 호출되고 계속 5초마다 반복적으로 실행될꺼에요~ 이 경우 memory maximum call stack size exceeded 가 발생할 가능성이 많습니다
 
그리고 `setTimeout`앞에 일반적으로 `window`객체는 생략합니다~ 

```suggestion
  useEffect(() => {
    timeoutRef.current = setTimeout(() => {
      setCurrent((prev) => (prev + 1) % images.length);
    }, 5000);

    return () => {
      if (timeoutRef.current) clearTimeout(timeoutRef.current);
    };
  }, []);
```